File: src/main/java/com/slavsquatsuperstar/mayonez/components/BoxCollider.java
package com.slavsquatsuperstar.mayonez.components;

import java.awt.Rectangle;

public class BoxCollider extends Component {

	private float x, y;
	private int width, height;

	public BoxCollider(int width, int height) {
		this.width = width;
		this.height = height;
	}

	@Override
	public void update(float dt) {
		x = parent.transform.position.x;
		y = parent.transform.position.y;
	}

	public Rectangle getBounds() {
		return new Rectangle((int) x, (int) y, width, height);
	}

}

File: mayonez-base/src/main/kotlin/mayonez/physics/colliders/BoxCollider.kt
package mayonez.physics.colliders

import mayonez.math.*
import mayonez.math.shapes.*

/**
 * An oriented rectangle defined by a width and height and centered around
 * the object's position.
 *
 * @constructor Constructs a box with the given dimensions
 * @author SlavSquatSuperstar
 */
open class BoxCollider(val size: Vec2) : PolygonCollider(Rectangle(Vec2(), size))